### PR TITLE
Harden MASM parsing and constant evaluation (lexer, repeats, constants, push slices)

### PR DESCRIPTION
**Pull request by huitseeker**
- Fix lexer span handling for invalid tokens after whitespace to avoid panics and return structured errors.
- Enforce repeat count bounds (including decorators) and document the max; add regression tests for zero/too-large counts.
- Enforce constant range checks during folding and evaluation, preventing modulus aliasing and overflows; add tests.
- Implement field division for / in constant folding to match docs; add tests.
- Validate push.WORD[...] slice indices with checked conversions and overflow guards; add single-index and range-index regression tests.

Per-commit review recommended.

### CODE REVIEW

File: crates/assembly-syntax/src/sema/passes/verify_repeat.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    analyzer: &'a mut AnalysisContext,
+}
+
+impl<'a> VerifyRepeatCounts<'a> {
+    pub fn new(analyzer: &'a mut AnalysisContext) -> Self {
+        Self { analyzer }
+    }
+}
+
+impl VisitMut for VerifyRepeatCounts<'_> {
```
Comment (bitwalker) [inline crates/assembly-syntax/src/sema/passes/verify_repeat.rs]: Since this pass doesn't mutate the AST, you can implement it as `Visit` instead - doesn't really matter too much, but better signals that the implementation is just analysis rather than transformation.

**Review by bitwalker — APPROVED** [review]

**Review by plafer — APPROVED** [review]

File: crates/assembly-syntax/src/ast/constants/eval.rs [hunk 2]
```diff
@@ -351,12 +353,24 @@ where
...
+                                    return Err(ConstEvalError::DivisionByZero {
+                                        span,
+                                        source_file: env.get_source_file_for(span),
+                                    }
+                                    .into());
+                                }
+                                let lhs = Felt::new(lhs.as_int());
+                                let rhs = Felt::new(rhs.as_int());
+                                IntValue::from(lhs / rhs)
+                            },
```
Comment (CergyK) [inline crates/assembly-syntax/src/ast/constants/eval.rs:373]: As discussed, this will be a confusing behavior (`/` as felt division, while `//` is the "usual" division).

Comment (huitseeker) [reply crates/assembly-syntax/src/ast/constants/eval.rs:373]: #3053
